### PR TITLE
Accounting oracle: tests & fixes

### PR DESCRIPTION
**Pull request by DiRaiks**
- [x] access control
- [x] initialize pre-conditions
- [x] legacy oracle migration
- [x] submitReportData checks contract version
- [x] submitReportData checks ref. slot
- [x] submitReportData checks data hash
- [x] submitReportData checks consensus version
- [x] submitReportData delivers the data to Lido and StakingRouter
- [x] submitReportData enforces extra data format
- [x] submitReportData enforces zero extra data items count for the empty format
- [x] submitReportData enforces zero data hash for the empty format
- [x] submitReportData enforces non-zero extra data items count for any format apart from empty
- [x] submitReportData enforces module ids sorting order
- [x] submitReportData enforces data safety boundaries
- [x] submitReportData resets and populates extra data processing state
- [x] submitReportExtraDataList enforces the deadline
- [x] submitReportExtraDataList checks extra data hash
- [x] submitReportExtraDataList checks ref slot
- [x] submitReportExtraDataList reverts if extra data already processed
- [x] (!!!) submitReportExtraDataList reverts if extra data partially processed [not reachable with current contract]
- [x] submitReportExtraDataList reverts if main data for the ref. slot not yet submitted
- [x] submitReportExtraDataList checks items count
- [x] submitReportExtraDataList enforces data format
- [x] submitReportExtraDataList enforces data sorting order
- [x] submitReportExtraDataList enforces data safety boundaries
- [x] submitReportExtraDataList delivers the data to staking router
- [x] submitReportExtraDataList updates extra data processing state
- [x] submitReportExtraDataList handles edge cases regarding the data
- [x] events (as part of other tests)
- [x] #634

### CODE REVIEW

**Review by skozin — APPROVED** [review]

File: contracts/0.8.9/oracle/ValidatorsExitBusOracle.sol [hunk 1]
```diff
@@ -147,11 +147,15 @@ contract ValidatorsExitBusOracle is BaseOracle, PausableUntil {
         ///
 
         /// @dev Total number of validator exit requests in this report. Must not be greater
-        /// than limit checked in OracleReportSanityChecker.checkExitBusOracleReport
+        /// than limit checked in OracleReportSanityChecker.checkExitBusOracleReport.
+        ///
+        /// Cannot be zero: in the case there's no newly exited or stuck validators at the moment
```
Comment (TheDZhon) [inline contracts/0.8.9/oracle/ValidatorsExitBusOracle.sol:152]: Didn't get the meaning of this comment.

AFAIK, we don't report exited and stuck validators from the ValidatorExitButOracle (using the accounting one instead).

Comment (skozin) [reply contracts/0.8.9/oracle/ValidatorsExitBusOracle.sol:152]: F, you're right. Will fix.

**Review by TheDZhon — APPROVED** [review]
It's huge :+1: :rocket:
